packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -7,13 +7,13 @@ import { getShadowTokenStyleSet } from './shadowStyle';
 export const Shadow = stagedComponent((props: ShadowProps) => {
   return (final: ShadowProps, children: React.ReactNode) => {
     const shadowTokenStyleSet = getShadowTokenStyleSet(props.shadowToken);
-    const mergedProps = mergeProps(final, { style: shadowTokenStyleSet.key });
+    const mergedProps = mergeProps(final, { style: shadowTokenStyleSet.ambient });
 
-    return (
-      <View style={shadowTokenStyleSet.ambient}>
-        <View {...mergedProps}>{children}</View>
-      </View>
-    );
+    const childrenArray = React.Children.toArray(children) as React.ReactElement[];
+    const child = childrenArray[0];
```

Comment:
Can we add some sort of check to child to enforce that is a <View> ? Also can we put it in the SPEC.md that Shadow _must_ take one child of type View?

Reply:
There is a React.Children.only() function which will check if there is only one child: [https://reactjs.org/docs/react-api.html#reactchildrenonly](https://reactjs.org/docs/react-api.html#reactchildrenonly)

In theory most children extend ViewProps so I don't think we need to necessarily enforce that the child is a View element? But I guess doing that is easier than trying to enforce that it has a style prop which can take shadow styling.

Reply:
In terms of checking for a specific type of child:

[https://stackoverflow.com/questions/27366077/only-allow-children-of-a-specific-type-in-a-react-component#:~:text=Use%20the%20React.Children.forEach%20method%20to%20iterate%20over%20the,%28%22Only%20card%20components%20allowed%20as%20children.%22%29%3B%20%7D%20%7D](https://stackoverflow.com/questions/27366077/only-allow-children-of-a-specific-type-in-a-react-component#:~:text=Use%20the%20React.Children.forEach%20method%20to%20iterate%20over%20the,%28%22Only%20card%20components%20allowed%20as%20children.%22%29%3B%20%7D%20%7D)

Reply:
Ideally we would be able to change the type of children to only be a single node or something instead of an Array but I haven't figured out how to do that with the stagedRender.

Reply:
I looked into React.Children.only, but my code kept triggering this error "React.Children.only expected to receive a single React element child" - I think because children is a ReactNode instead of a ReactElement. Didn't know if there was a good way to fix it, but I saw that in MenuTrigger, where I got some of the code for Shadow, we check the number of elements in the array and throw a console warning if there isn't exactly one. I copied that approach and also added a warning if the child isn't a view